contrib/ci/check-source.py

```diff
@@ -22,6 +22,8 @@ def test_files() -> int:
             continue
         if fn.startswith("contrib/ci"):
             continue
+        if fn.endswith("libfwupdplugin/fu-cab-firmware.c"):
```

Comment:
Why?

Reply:
Using g_byte_array_free_to_bytes() means we can just switch the header without copying the data, which means parsing a 1GB .cab file takes 1GB of RSS, not 2GB. It's basically saying "I know what I'm doing".

Reply:
Isn't it better to have a specific exception than quirk the whole file then?

Reply:
Yes, good idea -- I've added `/* nocheck */` -- other ideas welcome.

Reply:
that's good, make sure you add a documentation somewhere that it can be used too.

Reply:
I'll add it in the error message in check-source -- that feels like the most intuitive and useful place.